pkg/field/linked_field.go

```diff
@@ -0,0 +1,83 @@
+package field
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Fields are exposed as an interface instead of directly as pointer to this
+// struct. That enforces the immutability of fields outside this package. Once
+// a field is created, its contents cannot be changed or overwritten outside
+// this package by simply assigning one field to another, via
+// f1 = f2 or *fp1 = *fp2.
+type field struct {
+	key   string
+	value interface{}
+}
+
```

Comment:
[minor]
```suggestion
var _ fmt.Stringer = field{}
var _ Field = field{}

```

Reply:
This one makes sense:
`var` _ fmt.Stringer = field{}`

However, `var _ Field = field{}` doesn't.

See [interface checks](https://golang.org/doc/effective_go.html#blank_implements)

> Don't do this for every type that satisfies an interface, though. By convention, such declarations are only used when there are no static conversions already present in the code, which is a rare event.